apps/passwd.c

```diff
@@ -47,6 +33,14 @@ static unsigned const char cov_2char[64] = {
     0x73, 0x74, 0x75, 0x76, 0x77, 0x78, 0x79, 0x7A
 };
 
+static const char ascii_dollar[] = { 0x24, 0x00 };
+#if defined(CHARSET_EBCDIC) && !defined(CHARSET_EBCDIC_TEST)
+static const char ebcdic_dollar[] = { 0x5B, 0x00 };
+# define dollar ebcdic_dollar
+#else
+# define dollar ascii_dollar
+#endif
```

Comment:
There's exactly [one spot](https://github.com/openssl/openssl/pull/4229/files#diff-16eb4f58bda236c67d287baad522e84dR553) where that macro is used, and it fits this use case on an ASCII platform with `-DCHARSET_EBCDIC`:

```
./util/shlib_wrap.sh apps/openssl passwd -salt 'rounds=5000'`echo -n '$toolongsaltstring' | iconv -t ebcdicus` -5 `echo -n "This is just a test" | iconv -t ebcdicus` | iconv -f ebcdicus
```

Of course, I could just have a plain "$" instead of using that macro, and make the use case this instead:

```
./util/shlib_wrap.sh apps/openssl passwd -salt 'rounds=5000$'`echo -n 'toolongsaltstring' | iconv -t ebcdicus` -5 `echo -n "This is just a test" | iconv -t ebcdicus` | iconv -f ebcdicus
```

I think that's what I will do.  Less eye-brows raised, obviously...